org.uqbar.project.wollok/src/org/uqbar/project/wollok/model/WollokModelExtensions.xtend

```diff
@@ -435,4 +466,35 @@ class WollokModelExtensions {
 		val rootNodeName = e.name.trim
 		node.text.trim.equals(rootNodeName) && semanticsElements.contains(node.semanticElement.eClass.name) && doApplyRenameTo(e, node.semanticElement)
 	}
+	
+	def static dispatch expectsExpression(EObject e) { false }
```

Comment:
I'm not sure about this. If `expectsExpression` means that the argument expects expressions as children, I think that (a) there are many uncovered situations and (b) you should not look only at the parent object but also at the position, e.g. an if-construction allows statements for its branch-children but not for its condition child.